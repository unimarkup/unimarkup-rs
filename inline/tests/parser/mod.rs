use crate::snapshot::Snapshot;
use unimarkup_commons::test_runner::snap_test_runner::SnapTestRunner;
use unimarkup_inline::ParseInlines;

mod snapshot;

#[test]
fn test_parser_snapshots() {
    for case in crate::prepare_test_cases("spec/markup", "spec/snapshots/parser") {
        let runner = SnapTestRunner::with_fn(&case.name, case.input.as_str(), |symbols| {
            let rest: &[_] = &[];
            let inlines: Vec<_> = symbols.parse_inlines().collect();
            let snapshot = Snapshot::snap(&inlines[..]);
            (snapshot, rest)
        })
        .with_info(format!("Test '{}' from '{}'", case.name, case.file_name));

        unimarkup_commons::run_snap_test!(runner, &case.out_path);
    }
}
