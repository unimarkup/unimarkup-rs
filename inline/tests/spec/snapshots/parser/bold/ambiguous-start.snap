---
source: inline/tests/parser/mod.rs
info: "Test 'ambiguous-start' from 'markup\\bold.yml'"
---
Plain @ (1:1)->(1:10) (
    The next 
    ^^^^^^^^^
)
Italic @ (1:10)->(1:34) (
    Bold @ (1:11)->(1:19) (
        Plain @ (1:13)->(1:17) (
            word
            ^^^^
        )
    )
    Plain @ (1:19)->(1:34) (
         is bolditalic.
        ^^^^^^^^^^^^^^^
    )
)

---
With input:

The next ***word** is bolditalic.


